vtex/faststore
# feat: `Filter` with next/dynamic and lazy import

## What's the purpose of this pull request?

This PR involves significant changes to the filter components within the `ProductGallery` and related files. The main goal is to refactor the filter functionality by removing the `Filter` component and replacing it with `FilterDesktop` and `FilterSlider` components. Additionally, dynamic imports are introduced to optimize loading.

Key changes include:

Refactoring Filter Components:
* Removed the `Filter` component and its associated code from `Filter.tsx`.
* Updated `FilterDesktop.tsx` to include the GraphQL fragment previously in `Filter.tsx`.
* Added dynamic imports for `Filter`, `FilterFacetBoolean`, `FilterFacetBooleanItem`, `FilterFacetRange`, `FilterFacets`, and `FilterSlider` in `FilterSlider.tsx`.
* Modified `index.ts` to export `FilterDesktop` and `FilterSlider` instead of the removed `Filter` component.

Dynamic Imports:
* Replaced static imports with dynamic imports for `ProductCard`, `EmptyGallery`, `FilterDesktop`, and `FilterSlider` in `DefaultComponents.ts`.

## How to test it?

You can use the preview URL from the starter PR and double check that some JS are just loaded after Open the Filter in the mobile version.

https://sfj-d705510--starter.preview.vtex.app/office

https://github.com/user-attachments/assets/b830ad60-9496-4fe2-b2c4-2837bfc8b364

### Starters Deploy Preview

- https://github.com/vtex-sites/starter.store/pull/638
